mlflow/tracing/processor/otel.py

```diff
@@ -0,0 +1,64 @@
+import json
+import uuid
+from typing import Optional
+
+from opentelemetry.context import Context
+from opentelemetry.sdk.trace import ReadableSpan as OTelReadableSpan
+from opentelemetry.sdk.trace import Span as OTelSpan
+from opentelemetry.sdk.trace.export import BatchSpanProcessor, SpanExporter
+
+from mlflow.entities.trace_info import TraceInfo
+from mlflow.entities.trace_status import TraceStatus
+from mlflow.tracing.constant import TRACE_SCHEMA_VERSION, TRACE_SCHEMA_VERSION_KEY, SpanAttributeKey
+from mlflow.tracing.trace_manager import InMemoryTraceManager
+
+
+class OtelSpanProcessor(BatchSpanProcessor):
+    """
+    SpanProcessor implementation to export MLflow traces to a OpenTelemetry collector.
+
+    Extending OpenTelemetry BatchSpanProcessor to add some custom hooks to be executed when a span
+    is started or ended (before exporting).
+    """
+
+    def __init__(self, span_exporter: SpanExporter):
+        super().__init__(span_exporter)
+        self.span_exporter = span_exporter
+        self._trace_manager = InMemoryTraceManager.get_instance()
+
+    def on_start(self, span: OTelSpan, parent_context: Optional[Context] = None):
+        """
+        Handle the start of a span. This method is called when an OpenTelemetry span is started.
+
+        Args:
+            span: An OpenTelemetry Span object that is started.
+            parent_context: The context of the span. Note that this is only passed when the context
+                object is explicitly specified to OpenTelemetry start_span call. If the parent
+                span is obtained from the global context, it won't be passed here so we should not
+                rely on it.
+        """
+        # Generate a random request ID and trace info just for the sake of consistency
```

Comment:
If we have this configured to OTLP and are manually triggering tracing to record to the active Experiment within MLflow, these traces that will be sent to the collector will have differing request_id's. Is there a way to unify them, or does that incur a high level of complexity that isn't worth it?